spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/reload/Fabric8EventBasedSecretsChangeDetector.java

```diff
@@ -133,7 +134,12 @@ public void onAdd(Secret secret) {
 		public void onUpdate(Secret oldSecret, Secret newSecret) {
 			LOG.debug("Secret " + newSecret.getMetadata().getName() + " was updated in namespace "
 					+ newSecret.getMetadata().getNamespace());
-			onEvent(newSecret);
+			if (Objects.equals(oldSecret.getData(), newSecret.getData())) {
```

Comment:
we don't really care if anything but `data` has changed, so trigger a reload when data changes.

Reply:
Does the Fabric8 client not return secrets data as a byte array like K8S client does?